woocommerce/woocommerce-android
# [REST API] Check for user role before proceeding with Jetpack activation process

Closes: #8701 


### Description


REST API-based login allows users with admin or shop manager role to use the app.

However, the capability to activate (install + connect) Jetpack is only available for users with admin access.

Thus, this PR adds user role checking before letting activation happen. This PR also handles error cases, such as displaying an "admin role required" screen if needed.

The flow added in this PR looks like this:

![Screenshot 2023-04-05 at 14 32 17](https://user-images.githubusercontent.com/266376/230012343-ddb7af3d-d417-4342-b4c9-6a4042198ce7.png)


Apologize that the PR is rather large! Some of it are just compose UI code.

### Testing instructions


Prep: 
1. Start with a test .org site with no Jetpack,
2. Have two accounts in it, account (A) with Admin role, and account (B) with Site Manager role.
3. Make sure you're logged-in in wp-admin using account (A)

Testing:
1. Login to the app using account (B),
2. Tap Jetpack banner, then on the benefits dialog tap the "Log in to Continue" button,
3. After fetching message is shown, make sure you are moved to the Jetpack Activation User Eligibility Error screen
4. Test the help and learn more buttons,
5. Test the retry button (after fetching, it should stay showing the error screen),
6. Go back to wp-admin, change user (B)'s role to administrator,
7. Test the retry button again. After fetching it should show the "Install Jetpack" screen which is the start of the activation flow.

### Images/gif


User role eligibility error screen:

<img src="https://user-images.githubusercontent.com/266376/230013033-4f217309-146b-4710-bd94-17b3c2dda733.png" width="42%" />
